src/Microsoft.ML.Parquet/ParquetLoader.cs

```diff
@@ -390,9 +390,21 @@ public Cursor(ParquetLoader parent, Func<int, bool> predicate, IRandom rand)
                     Columns = _loader._columnsLoaded.Select(i => i.Name).ToArray()
                 };
 
-                int numBlocks = (int)Math.Ceiling(((decimal)parent.GetRowCount() / _readerOptions.Count));
-                int[] blockOrder = _rand == null ? Utils.GetIdentityPermutation(numBlocks) : Utils.GetRandomPermutation(rand, numBlocks);
-                _blockEnumerator = blockOrder.GetEnumerator();
+                try
+                {
+                    int numBlocks = checked((int)Math.Ceiling(((decimal)parent.GetRowCount() / _readerOptions.Count)));
+                    int[] blockOrder = _rand == null ? Utils.GetIdentityPermutation(numBlocks) : Utils.GetRandomPermutation(rand, numBlocks);
+                    _blockEnumerator = blockOrder.GetEnumerator();
+                }
+                catch (Exception e)
+                {
+                    if (e is OutOfMemoryException || e is OverflowException)
+                    {
+                        throw new InvalidDataException("Error due to too many blocks. Try increasing block size.", e);
```

Comment:
>, e [](start = 113, length = 3)

Given that we're interpreting the exception pretty exactly, I'm not certain that there is much utility in including that inner exception. My reason is: usually in this API and many APIs I've seen, innermost exceptions are usually the "real" problem and practically the only thing people wind up looking at. :) So it would be nice if your nice friendly error message you have written was that innermost exception, and I think you have a very good excuse to make it the innermost exception. :)